ci/test_cpp.sh

```diff
@@ -9,7 +9,23 @@ rapids-logger "Generate C++ testing dependencies"
 rapids-dependency-file-generator \
   --output conda \
   --file_key test_cpp \
-  --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch)" | tee env.yaml
+  --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch)" | tee test.yaml
+
+rapids-logger "Downloading artifacts from previous jobs"
+CPP_CHANNEL=$(rapids-download-conda-from-s3 cpp)
+
+cat <<EOF | tee cuml.yaml
+name: cuml
+channels:
+  - ${CPP_CHANNEL}
+dependencies:
+  - libcuml
+  - libcuml-tests
```

Comment:
Assuming that conda-merge will place `CPP_CHANNEL` at a sufficient priority so that we grab build artifacts, but do we lose anything also constraining the packages to that channel?

```suggestion
  - ${CPP_CHANNEL}::libcuml
  - ${CPP_CHANNEL}::libcuml-tests
```

Reply:
This is a good question. I designed https://github.com/rapidsai/dependency-file-generator/pull/67 in such a way that the CI artifact channels will always be prepended. It's still possible that another channel could be used, but there's not a good way to inject the channel names corresponding to those packages if we go with the route of https://github.com/rapidsai/dependency-file-generator/pull/67.